file/delete_scheduler.h

```diff
@@ -45,6 +45,7 @@ class DeleteScheduler {
   // Set delete rate limit in bytes per second
   void SetRateBytesPerSecond(int64_t bytes_per_sec) {
     rate_bytes_per_sec_.store(bytes_per_sec);
+    CreateBackgroundThread();
```

Comment:
How about renaming to `MaybeCreateBackgroundThread()` in which you check the condition and create the thread if necessary? This function can also be called in the constructor, and the check outside this function can be removed.